src/riakc_obj.erl

```diff
@@ -494,6 +498,26 @@ add_secondary_index(MD, [{Id, BinList} | Rest]) when is_binary(Id) ->
             add_secondary_index(MD2, Rest)
     end.
 
+%% @doc  Clear TTL entry
+-spec clear_ttl(metadata()) -> metadata().
+clear_ttl(MD) ->
+    dict:erase(?MD_TTL, MD).
+
+%% @doc  Get TTL
+-spec get_ttl(metadata()) -> ttl() | undefined.
+get_ttl(MD) ->
+    case dict:find(?MD_TTL, MD) of
+        {ok, TTL} ->
+            TTL;
+        error ->
+            false
```

Comment:
Okay, so it seems this is one of those cases where dialyzer doesn't yell at you unless you add certain non-default warning flags that are prone to yielding false positives. If you add the `-Wspecdiffs` flag, it will tell you:
```
Type specification riakc_obj:get_ttl(metadata()) -> ttl() | 'undefined' is a subtype of the success typing: riakc_obj:get_ttl(dict()) -> any()
```

Which is technically true: that dict we pass in could contain anything, so technically that function could return `undefined`, just like the spec says.

On the other hand, I'm not sure why it doesn't complain about the clause that returns `false` being outside the type spec. I suspect that dialyzer may optimistically assume that we'll never call that function in a way that hits the `error` case, unless it can specifically find a place where we force it to take that branch. Then again, I'm not sure if it can verify that, since I don't know if the `dict` API specs are specific enough for it to determine that in any scenario.

I can at least get it to throw an error by adding the line `false = get_ttl(dict:new())` elsewhere in the module, in which case it complains:
```
The pattern 'false' can never match the type 'undefined' | non_neg_integer()
```